apps/advisor/pkg/app/utils_test.go

```diff
@@ -136,10 +173,14 @@ type mockStep struct {
 	err error
 }
 
-func (m *mockStep) Run(ctx context.Context, obj *advisorv0alpha1.CheckSpec, items []any) ([]advisorv0alpha1.CheckReportError, error) {
+func (m *mockStep) Run(ctx context.Context, obj *advisorv0alpha1.CheckSpec, items any) (*advisorv0alpha1.CheckReportError, error) {
```

Comment:
This function confused me a bit. Rather than passing a string as an error, maybe pass an actual error on line 96 instead. Then can do:

```suggestion
func (m *mockStep) Run(ctx context.Context, obj *advisorv0alpha1.CheckSpec, items any) (*advisorv0alpha1.CheckReportError, error) {
	if m.err != nil {
		return nil, m.err
	}
	if _, ok := items.(error); !ok {
		return &advisorv0alpha1.CheckReportError{}, nil
	}
	return nil, nil
}
```

Reply:
this defeats the purpose that items are always the same type but I guess that for the test it can be more readable.